src/Controls/samples/Controls.Sample/Controls/RedButton/RedButton.cs

```diff
@@ -0,0 +1,15 @@
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Hosting;
+
+namespace Maui.Controls.Sample.Controls
+{
+	public class RedButton : Button
+	{
+		static RedButton()
+		{
+			var handlers = AppHost.Current.Handlers.GetCollection();
+
+			handlers.AddHandler<RedButton, RedButtonHandler>();
```

Comment:
This is the way I would think to register new handlers. Instead of using the services which may be scoped, using the static `AppHost` will always be the root and any scoped services may override this.